feature_engine/discretisation/base_discretiser.py

```diff
@@ -0,0 +1,120 @@
+# Authors: Morgan Sell <morganpsell@gmail.com>
+# License: BSD 3 clause
+
+from typing import List, Optional, Union
+
+import pandas as pd
+
+from sklearn.utils.validation import check_is_fitted
+
+from feature_engine.dataframe_checks import (
+    _check_contains_na,
+    _check_input_matches_training_df,
+    _is_dataframe,
+)
+
+from feature_engine.base_transformers import BaseNumericalTransformer
+from feature_engine.dataframe_checks import _check_contains_na
+
+import warnings
+
+
+class BaseDiscretiser(BaseNumericalTransformer):
+    """
+    Share set-up checks and methods across numerical discretizers
+
+    Parameters
+    ----------
+    return_object: bool, default=False
+        Whether the the discrete variable should be returned as numeric or as
+        object. If you would like to proceed with the engineering of the variable as if
+        it was categorical, use True. Alternatively, keep the default to False.
+
+    return_boundaries : bool, default=False
+        Whether the output should be the interval boundaries. If True, it returns
+        the interval boundaries. If False, it returns integers.
+
+
+    """
+
+    def __init__(
+            self,
+            return_object: bool = False,
+            return_boundaries: bool = False,
+    ) -> None:
+
+        if not isinstance(return_object, bool):
+            raise ValueError("return_object must be True or False")
+
+        if not isinstance(return_boundaries, bool):
+            raise ValueError("return_boundaries must be True or False")
+
+        self.return_object = return_object
+        self.return_boundaries = return_boundaries
+
+    def _check_transform_input_and_state(self, X: pd.DataFrame) -> pd.DataFrame:
```

Comment:
We don't need this here. We are importing this functionality from the fit method from the BaseNumericalTransformer.

I know that it looks different from the BaseCategorical, but this was a design error if you want, that is too big to fix in this PR. 

Can you pls remove this function?